api/v1beta1/vmagent_types.go

```diff
@@ -202,6 +202,11 @@ type VMAgentSpec struct {
 	// +optional
 	ProbeNamespaceSelector *metav1.LabelSelector `json:"probeNamespaceSelector,omitempty"`
 
+	// NodeScrapeSelector defines VMNodeScrape to be selected for scraping.
+	// if neither PodScrapeNamespaceSelector, nor ProbeSelector nor PodScrapeSelector, nor NodeScrapeSelector are specified,
+	// configuration is unmanaged.
+	// +optional
+	NodeScrapeSelector *metav1.LabelSelector `json:"nodeScrapeSelector,omitempty"`
```

Comment:
It would be nice if `nodeScrapeNamespaceSelector` is also added. `VMNodeScrapeSpec` doesn't need `namespaceSelector` because Nodes themselves are not namespaced. But VMNodeScrapes are namespaced like VMServiceScrapes, so `VMAgentSpec` needs `nodeScrapeNamespaceSelector`

Reply:
I thought about it, but maybe its too complicated for user? You have to care about additional selector and its correctness. 
As far as i know, prometheus made NamespaceSelectors deprecated. 

 Any way, its not hard to add, but its hard to remove later)

Reply:
There are already namespace selectors for other scraping CRs (`VMServiceScrape` etc.), so I don't think complexity will be a problem. It is preferable to have it for consistency.

I read the [API documentation for the Prometheus operator](https://github.com/prometheus-operator/prometheus-operator/blob/v0.44.1/Documentation/api.md), but I couldn't find any mention of deprecating namespace selectors. Instead, it says `serviceMonitorSelector` and `podMonitorSelector` are deprecated. (It makes me wonder).

Reply:
Indeed, thanks for pointing, i've thought, that namespaceSelectors are deprecated.. 

 Added implementation for nodeScrapeNamespaceSelector.